float32 acc
float32 delta